File: matilda/messages/message_server.py
from maddie.dependency import Dependency
from maddie.dependency_container import DependencyContainer

from matilda.messages.handler.message_dispatcher import MessageDispatcher
from matilda.messages.message_receiver import MessageReceiver


class MessageServer(Dependency):
    def __init__(self, message_receiver: MessageReceiver, message_dispatcher: MessageDispatcher):
        self.__message_receiver = message_receiver
        self.__message_dispatcher = message_dispatcher

    def start(self):
        while True:
            try:
                message = self.__message_receiver.receive()
                self.__message_dispatcher.dispatch(message)
            except EOFError:
                return

    @staticmethod
    def create(dependency_container: DependencyContainer) -> 'MessageServer':
        return MessageServer(dependency_container.get(MessageReceiver),
                             dependency_container.get(MessageDispatcher))

File: agent/library/src/main/java/org/matilda/messages/MessageServer.kt
package org.matilda.messages

import kotlinx.io.EOFException
import me.tatarka.inject.annotations.Inject
import org.matilda.messages.handlers.MessageHandler

@Inject
class MessageServer(private val mMessageReceiver: MessageReceiver, private val mMessageHandler: MessageHandler) {
    suspend fun start() {
        while (true) {
            try {
                val message = mMessageReceiver.receive()
                mMessageHandler.handle(message)
            } catch (ignored: EOFException) {
                return
            }
        }
    }
}
